smdebug/tensorflow/base_hook.py

```diff
@@ -183,6 +183,17 @@ def _get_worker_name(self) -> str:
         """
         self._assert_distribution_strategy()
         if self.distribution_strategy == TFDistributionStrategy.HOROVOD:
+            try:
+                # when model parallel is being used, there will be multiple processes
+                # with same hvd rank, hence use smp.rank
+                import smdistributed.modelparallel.tensorflow as smp
+
+                if smp.core.initialized:
+                    # if smp is in use
+                    return f"worker_{smp.rank()}"
```

Comment:
Can this block be entered more than once? If yes, I would recommend to  cache this if possible. We have seen calling import like this causes perf penalty. So ensure that the import is not called more than once.

Reply:
Modified it